File: 0078-subsets/0078-subsets.java
class Solution {

    public static void subsetfunc(int[] nums,int index,List<Integer>sublist,List<List<Integer>>ans)
    
    {
        ans.add(new ArrayList<>(sublist));

        for(int i = index; i < nums.length; i++)
        {
            sublist.add(nums[i]);
            subsetfunc(nums,i+1,sublist,ans);
            sublist.remove(sublist.size()-1);
        }
    }
    public List<List<Integer>> subsets(int[] nums) {
        List<List<Integer>>ans = new ArrayList<>();
        subsetfunc(nums,0,new ArrayList<>(),ans);
        return ans;
        
    }
}

File: 0078-subsets/0078-subsets.cpp
class Solution {
public:
    vector<vector<int>> ans;
    vector<int> subset;

    void solve(vector<int>& nums, int index) {

        ans.push_back(subset);

        for (int i = index; i < nums.size(); i++) {

          
            subset.push_back(nums[i]);

           
            solve(nums, i + 1);

            subset.pop_back();
        }
    }

    vector<vector<int>> subsets(vector<int>& nums) {
        solve(nums, 0);
        return ans;
    }
};